lib/AST/ClangTypeConverter.cpp

```diff
@@ -158,6 +158,57 @@ const clang::Type *ClangTypeConverter::getFunctionType(
   llvm_unreachable("invalid representation");
 }
 
+const clang::Type *ClangTypeConverter::getFunctionType(
+    ArrayRef<SILParameterInfo> params, Optional<SILResultInfo> result,
+    SILFunctionType::Representation repr) {
+
+  // Using the interface type is sufficient as Swift does not allow abstracting
+  // over @convention(c) functions, hence we don't need any substitutions.
```

Comment:
You might want to codify this with an assertion:

`assert(!result.getValue().getInterfaceType()->hasTypeParameter())`

Reply:
Fixed.

Reply:
So... this actually caught something. Sometimes we end up getting a generic type `τ_0_0`, specifically when compiling `test/SILGen/objc_imported_generic.swift`. At the moment, that seems to be getting converted to
```
ObjCObjectPointerType 0x7fce4c822e40 'id' imported
`-ObjCObjectType 0x7fce4c822e10 'id' imported
```
in `GenClangType.cpp`. The new `ClangTypeConverter.cpp` code also has the same behavior (since it is essentially the same as `GenClangType.cpp`). So, I think I should remove this assertion and just let it be?

Reply:
Turns out, this is not true even for `@convention(c)` functions. Specifically, the block invocation function can have a generic parameter in its return type. I will update the comment.